src/Features/Core/Portable/AddParameter/AbstractAddParameterCodeFixProvider.cs

```diff
@@ -113,125 +137,269 @@ internal abstract class AbstractAddParameterCodeFixProvider<
             }
 
+            return methodsAndArgumentToAdd.ToImmutableAndFree();
+        }
+
+        private int NonParamsParameterCount(IMethodSymbol method)
+            => method.IsParams() ? method.Parameters.Length - 1 : method.Parameters.Length;
+
+        private void RegisterFixForMethodOverloads(
+            CodeFixContext context,
+            SeparatedSyntaxList<TArgumentSyntax> arguments,
+            ImmutableArray<ArgumentInsertPositionData<TArgumentSyntax>> methodsAndArgumentsToAdd)
+        {
             // Order by the furthest argument index to the nearest argument index.  The ones with
             // larger argument indexes mean that we matched more earlier arguments (and thus are
             // likely to be the correct match).
-            foreach (var tuple in constructorsAndArgumentToAdd.OrderByDescending(t => t.index))
+            foreach (var argumentInsertPositionData in methodsAndArgumentsToAdd.OrderByDescending(t => t.ArgumentInsertionIndex))
             {
-                var constructor = tuple.constructor;
-                var argumentToAdd = tuple.argument;
+                var methodToUpdate = argumentInsertPositionData.MethodToUpdate;
+                var argumentToInsert = argumentInsertPositionData.ArgumentToInsert;
+                var parameters = methodToUpdate.Parameters.Select(p => p.ToDisplayString(SimpleFormat));
+                var title = GetCodeFixTitle(FeaturesResources.Add_parameter_to_0, methodToUpdate, parameters);
+                var hasCascadingDeclarations = HasCascadingDeclarations(methodToUpdate);
 
-                var parameters = constructor.Parameters.Select(p => p.ToDisplayString(SimpleFormat));
-                var signature = $"{type.Name}({string.Join(", ", parameters)})";
+                if (hasCascadingDeclarations)
+                {
+                    var titleForCascadingFix = GetCodeFixTitle(FeaturesResources.Add_parameter_to_0_including_overrides_implementations, methodToUpdate, parameters);
+                    context.RegisterCodeFix(new CodeAction.CodeActionWithNestedActions(title, ImmutableArray.Create<CodeAction>(
+                            new MyCodeAction(title,
+                                c => FixAsync(context.Document, methodToUpdate, argumentToInsert, arguments, fixAllReferences: false, c)),
+                            new MyCodeAction(titleForCascadingFix,
+                                c => FixAsync(context.Document, methodToUpdate, argumentToInsert, arguments, fixAllReferences: true, c))),
+                            isInlinable: true),
+                        context.Diagnostics);
+                }
+                else
+                {
+                    context.RegisterCodeFix(
+                        new MyCodeAction(title,
+                            c => FixAsync(context.Document, methodToUpdate, argumentToInsert, arguments, fixAllReferences: false, c)),
+                        context.Diagnostics);
+                }
```

Comment:
it's minor, but something about the duplication here feels bad ot me.  can you try:

```c#
// pseudo code.
var action = new MyCodeAction(..., fixAllReferences: false, ...);
if (hasCascading) {
    action = new CodeActionWithNested(...
            action, 
            new MyCodeAction(... fixAllReferences: true, ...);
}

RegisterCodeFix(action);
```
?